### PR TITLE
Integrating logs storage for Tuna through Logstash and Elastic Search

### PR DESCRIPTION
**Pull request by reidkwja**
This PR allows the Tuna codebase to integrate with logstash and elastic search infrastructure for logs storage and analysis.

### CODE REVIEW

File: tuna/utils/logger.py [hunk 1]
```diff
@@ -27,10 +27,16 @@
...
 from typing import Union
+from logstash_async.handler import AsynchronousLogstashHandler
+from logstash_async.handler import LogstashFormatter
 from tuna.utils.metadata import TUNA_LOG_DIR
 
+logstash_status = os.getenv('TUNA_LOGSTASH_STATUS', 'false').lower() == 'true'
+logstash_host = os.getenv('TUNA_LOGSTASH_HOST', 'localhost')
+logstash_port = int(os.getenv('TUNA_LOGSTASH_PORT', "5000"))
+logstash_path = os.getenv('TUNA_LOGSTASH_PATH', None)
+
```
Comment (alexandraBara) [inline tuna/utils/logger.py]: they dont need to be globals, you can write a getter function that returns the logstash handler and call it both inside setup_logger and set_usr_logger. Sample:
```
def get_logstash_logger():
  logstash_handler: AsynchronousLogstashHandler = AsynchronousLogstashHandler(
      host=logstash_host, port=logstash_port, database_path=logstash_path)
    logstash_formatter: LogstashFormatter = LogstashFormatter()
    logstash_handler.setFormatter(logstash_formatter)
  logstash_handler.setLevel(logging.INFO)
```
then move the os.getenvs inside the functions and something like:
```
if logstash_status:
  logger.addHandler(get_logstash__logger())
```

**Review by alexandraBara — CHANGES_REQUESTED** [review]

**Review by alexandraBara — APPROVED** [review]
looks good

File: tuna/utils/logger.py [hunk 2]
```diff
@@ -51,12 +63,24 @@ def setup_logger(logger_name: str = 'Tuna',
...
 
+  logstash_status, logstash_host, logstash_port, logstash_path = get_logstash_config(
+  )
+
+  if logstash_status:
+    logstash_handler: AsynchronousLogstashHandler = AsynchronousLogstashHandler(
+        host=logstash_host, port=logstash_port, database_path=logstash_path)
+    logstash_formatter: LogstashFormatter = LogstashFormatter()
+    logstash_handler.setFormatter(logstash_formatter)
+    logstash_handler.setLevel(logging.INFO)
```
Comment (cderb) [inline tuna/utils/logger.py]: This could be made into a function. Which could return the handler + status.

File: tuna/utils/logger.py [hunk 3]
```diff
@@ -77,5 +101,17 @@ def set_usr_logger(logger_name: str) -> logging.Logger:
...
+
+  logstash_status, logstash_host, logstash_port, logstash_path = get_logstash_config(
+  )
+
+  if logstash_status:
+    logstash_handler: AsynchronousLogstashHandler = AsynchronousLogstashHandler(
+        host=logstash_host, port=logstash_port, database_path=logstash_path)
+    logstash_formatter: LogstashFormatter = LogstashFormatter()
+    logstash_handler.setFormatter(logstash_formatter)
+    logstash_handler.setLevel(logging.INFO)
```
Comment (cderb) [inline tuna/utils/logger.py]: Could re-use function here.

**Review by cderb — CHANGES_REQUESTED** [review]

File: tuna/utils/logger.py [hunk 4]
```diff
@@ -74,12 +84,11 @@ def setup_logger(logger_name: str = 'Tuna',
...
-        host=logstash_host, port=logstash_port, database_path=logstash_path)
-    logstash_formatter: LogstashFormatter = LogstashFormatter()
-    logstash_handler.setFormatter(logstash_formatter)
-    logstash_handler.setLevel(logging.INFO)
-    logger.addHandler(logstash_handler)
+    add_logstash_handler(logger, logstash_host, logstash_port, logstash_path)
+    logger.info("Logstash is enabled. Sending logs to %s:%d", logstash_host,
+                logstash_port)
+  else:
+    logger.info("Logstash is disabled. Not sending logs to Logstash.")
```
Comment (alexandraBara) [inline tuna/utils/logger.py]: we dont need this printout, only print when enabled is enough

File: tuna/utils/logger.py [hunk 5]
```diff
@@ -106,12 +115,11 @@ def set_usr_logger(logger_name: str) -> logging.Logger:
...
-        host=logstash_host, port=logstash_port, database_path=logstash_path)
-    logstash_formatter: LogstashFormatter = LogstashFormatter()
-    logstash_handler.setFormatter(logstash_formatter)
-    logstash_handler.setLevel(logging.INFO)
-    lgr.addHandler(logstash_handler)
+    add_logstash_handler(lgr, logstash_host, logstash_port, logstash_path)
+    lgr.info("Logstash is enabled. Sending logs to %s:%d", logstash_host,
+             logstash_port)
+  else:
+    lgr.info("Logstash is disabled. Not sending logs to Logstash.")
```
Comment (alexandraBara) [inline tuna/utils/logger.py]: same here

**Review by cderb — APPROVED** [review]

**Pull request merged**